src/peer_digest.cc

```diff
@@ -65,19 +66,10 @@ static const time_t GlobDigestReqMinGap = 1 * 60;   /* seconds */
 
 static time_t pd_last_req_time = 0; /* last call to Check */
 
-PeerDigest::PeerDigest(CachePeer * p)
+PeerDigest::PeerDigest(CachePeer * const p):
+    peer(p),
+    host(peer->host) // if peer disappears, we will know its name
 {
-    assert(p);
```

Comment:
This may seem like an assertion removal (and a possibly nil `peer` pointer dereference above), but the new `peer->...` call above asserts when `p` is nil, so the safety check in the official code are preserved by this change.